### PR TITLE
Upgrade to `mini_racer` gem

### LINKED ISSUE #1790: Replace abandoned 'therubyracer' gem / XML device message parser
The `therubyracer` gem has been abandoned for years, and uses a very old version of V8. We use it compile assets, so we can replace it with anything, but it appears we also use it to run arbitrary code to parse manifest fields (see `ManifestFieldMapping`).

### PR DESCRIPTION
**Pull request by ysbaddaden**
closes #1790 

I implemented a custom XML serializer to expose a trivial data structure to the JavaScript context because existing accessible solutions (e.g. Hash.from_xml()` from ActiveSupport) weren't very reliable (e.g. override attributes with contents), or required to introduce yet another dependency to handle something quite easy to implement in the end, and where we're in full control of the generated data structure (e.g. we can move to another XML library).

### CODE REVIEW

File: spec/models/manifest_operations_spec.rb
Comment (ysbaddaden) [inline spec/models/manifest_operations_spec.rb:136]: For example this XML document is serialized as:

```json
{
  "name": "Message",
  "children": [
    {
      "name": "Patient",
      "attributes": [{"name": "Socrates", "age": "17"}]
    }
  ]
}
```

**Comment by diegoliberman** [comment]
Let's wait until January to merge this, so that we send it to production when Julien is back from vacations.

**Comment by leandroradusky** [comment]
Looks good to me :)

**Review by leandroradusky — APPROVED** [review]

**Pull request merged**
